optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -53,6 +54,8 @@
     private BenchmarkReport benchmarkReport = null;
 
     private ExecutorService executorService;
+    private ExecutorService warmUpExecutorService;
+    private ExecutorCompletionService<SingleBenchmarkRunner> warmUpExecutorCompletionService;
```

Comment:
trivial: I'd order executorService after the 2 warmup ones, because it used later (and has the same volatility).

Elaborate philosophy:
I tend to organize fields, first on volatility (static final, static, final instance field, rarely-changing instance field, consistently-changing) and then on usage order (= some sort of logical order).
I'd even say there is logical order between things that are equal in importance. For example: in DefaultSolver, the Termination is logically before the BestSolutionRecaller field. This means that phaseStarted(), stepEnded() and similar events will be called in that order: first on Termination, then on BestSolutionRecaller. So if in the future, BestSolutionRecaller need something from Termination, it can safely presume it's already initialized & up to date. If on the other hand Termination needs something from BestSolutionRecaller, they need to be reserved in field & method call order (which implying checking if there's no dependency in the other direction).

Having a consistent ordering can actually prevents bugs in rare occasions, such as deadlocks. For example:Thread 1 takes A then takes B and thread 2 takes B then takes A: if they both start with A this is impossible.